File: app/src/main/java/ru/nsu/fit/modao/fragments/CreateGroupFragment.java
package ru.nsu.fit.modao.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import ru.nsu.fit.modao.R;
import ru.nsu.fit.modao.databinding.FragmentCreateGroupBinding;
import ru.nsu.fit.modao.model.Group;
import ru.nsu.fit.modao.model.ShortInfoGroup;
import ru.nsu.fit.modao.repository.MainViewModel;
import ru.nsu.fit.modao.repository.MyApplication;

public class CreateGroupFragment extends Fragment {
    FragmentCreateGroupBinding binding;
    MainViewModel mainViewModel;
    MyApplication app;
    boolean request = true;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentCreateGroupBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mainViewModel = new ViewModelProvider(getActivity()).get(MainViewModel.class);
        app = (MyApplication) getActivity().getApplication();
        mainViewModel.getNewGroup().observe(getViewLifecycleOwner(), group -> {
            if (request){
                return;
            }
            if (group.getId() == -1L){
                binding.tipMessage.setText(R.string.incorrectData);
            }
            else if (group.getId() == -2L){
                binding.tipMessage.setText(R.string.serverProblems);
            }
            else {
                ShortInfoGroup shortGroup = new ShortInfoGroup(group.getGroupName(), group.getId());
                NavDirections action = CreateGroupFragmentDirections.actionCreateGroupFragmentToGroupInfoFragment(shortGroup);
                Navigation.findNavController(view).navigate(action);
            }
        });
        binding.buttonNext.setOnClickListener(v -> {
            request = false;
            String name = binding.nameText.getText().toString();
            String description = binding.descriptionText.getText().toString();
            if (name.equals("")){
                binding.tipMessage.setText(R.string.enterData);
                return;
            }
            mainViewModel.createGroup(app.getQuestApi(),
                    new Group(name, null, description, 0), app.getUserID());
        });
    }
}

File: app/src/main/java/ru/nsu/fit/modao/fragments/CreateGroupFragment.kt
package ru.nsu.fit.modao.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import dagger.hilt.android.AndroidEntryPoint
import ru.nsu.fit.modao.R
import ru.nsu.fit.modao.databinding.FragmentCreateGroupBinding
import ru.nsu.fit.modao.models.Group
import ru.nsu.fit.modao.viewmodels.MainViewModel
@AndroidEntryPoint
class CreateGroupFragment: Fragment() {
    private var _binding: FragmentCreateGroupBinding? = null
    private val binding get() = _binding!!
    private var group: Group? = null
    private val mainViewModel: MainViewModel by viewModels()
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentCreateGroupBinding.inflate(inflater, container, false)
        return binding.root

    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        mainViewModel.groupId.observe(viewLifecycleOwner) {
            mainViewModel.getGroupInfo(it)
        }
        mainViewModel.groupInfo.observe(viewLifecycleOwner) {
            findNavController().navigate(CreateGroupFragmentDirections
                .actionCreateGroupFragmentToGroupInfoFragment(it))
        }
        binding.buttonNext.setOnClickListener {
            val name = binding.nameText.text.toString()
            if (name == ""){
                binding.tipMessage.setText(R.string.enterData)
                return@setOnClickListener
            }
            group = Group(typeGroup = 0, groupName = name,
                description = binding.descriptionText.text.toString())
            mainViewModel.createGroup(group!!)
        }
    }
}